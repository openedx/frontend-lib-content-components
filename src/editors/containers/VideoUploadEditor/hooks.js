import * as module from './hooks';
import { selectors, thunkActions } from '../../data/redux';
import store from '../../data/store';
import * as appHooks from '../../hooks';

export const {
  navigateTo,
} = appHooks;

export const postUploadRedirect = (storeState) => {
  const learningContextId = selectors.app.learningContextId(storeState);
  const blockId = selectors.app.blockId(storeState);
  return (videoUrl) => navigateTo(`/course/${learningContextId}/editor/video/${blockId}?selectedVideoUrl=${videoUrl}`);
};

export const onVideoUpload = () => {
  const storeState = store.getState();
  return module.postUploadRedirect(storeState);
};

export const useUploadVideo = async ({
  dispatch,
  supportedFiles,
  setLoadSpinner,
  postUploadRedirectFunction,
}) => {
  dispatch(thunkActions.video.uploadVideo({
    supportedFiles,
    setLoadSpinner,
    postUploadRedirectFunction,
  }));
};

export const useHistoryGoBack = () => (() => window.history.back());

export default {
  postUploadRedirect,
  onVideoUpload,
  useUploadVideo,
  useHistoryGoBack,
};
