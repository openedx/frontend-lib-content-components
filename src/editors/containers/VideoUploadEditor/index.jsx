import React from 'react';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Spinner } from '@edx/paragon';
import './index.scss';
import messages from './messages';
import { VideoUploader } from './VideoUploader';

export const VideoUploadEditor = () => {
  const [loading, setLoading] = React.useState(false);
  const intl = useIntl();

  return (!loading) ? (
    <div className="d-flex marked-area flex-column p-3">
      <VideoUploader setLoading={setLoading} />
    </div>
  ) : (
    <div style={{
      position: 'absolute',
      left: '50%',
      top: '50%',
      transform: 'translate(-50%, -50%)',
    }}
    >
      <Spinner
        animation="border"
        className="m-3"
        screenreadertext={intl.formatMessage(messages.spinnerScreenReaderText)}
      />
    </div>
  );
};

export default VideoUploadEditor;
