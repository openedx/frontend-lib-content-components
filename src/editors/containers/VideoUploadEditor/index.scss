@import "@edx/paragon/scss/core/core";

.dropzone-middle {
  border: 2px dashed #ccc;

  &.active {
    border: 2px solid #262626; /* change color when active */
  }
}

.pgn__dropzone {
  height: 96vh;
  width: 100%;
}

.video-id-prompt {

  input::placeholder {
    color: #454545;
    // color: #5E35B1;
    font-weight: '500';
    word-wrap: 'break-word';
  }

  button {
    border: none !important;
    background-color: #FFFFFF;
  }

  .btn-icon.url-submit-button {
    &, &:active, &:hover {
      background-color: transparent !important;
      border: none !important;
      color: $gray-500 !important;
    }
  }

  .prompt-button {
    background: rgba(239, 234, 247, 0.70);
  }
}
