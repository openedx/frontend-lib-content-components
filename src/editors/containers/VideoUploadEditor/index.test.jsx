import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { initializeMockApp } from '@edx/frontend-platform';
import { AppProvider } from '@edx/frontend-platform/react';
import { configureStore } from '@reduxjs/toolkit';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import VideoUploadEditor from '.';

jest.unmock('react-redux');
jest.unmock('@edx/frontend-platform/i18n');
jest.unmock('@edx/paragon');
jest.unmock('@edx/paragon/icons');

describe('VideoUploadEditor', () => {
  let store;

  const renderComponent = async (storeParam) => render(
    <AppProvider store={storeParam}>
      <IntlProvider locale="en">
        <VideoUploadEditor />
      </IntlProvider>,
    </AppProvider>,
  );

  beforeEach(async () => {
    store = configureStore({
      reducer: (state, action) => ((action && action.newState) ? action.newState : state),
      preloadedState: {},
    });

    initializeMockApp({
      authenticatedUser: {
        userId: 3,
        username: 'test-user',
        administrator: true,
        roles: [],
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders as expected with default behavior', async () => {
    expect(await renderComponent(store)).toMatchSnapshot();
  });

  it('calls window.history.back when close button is clicked', async () => {
    const container = await renderComponent(store);
    const closeButton = container.getAllByRole('button', { name: /close/i });
    const oldHistoryBack = window.history.back;
    window.history.back = jest.fn();

    expect(closeButton).toHaveLength(1);
    expect(window.history.back).not.toHaveBeenCalled();
    closeButton.forEach((button) => fireEvent.click(button));
    expect(window.history.back).toHaveBeenCalled();

    window.history.back = oldHistoryBack;
  });
});
