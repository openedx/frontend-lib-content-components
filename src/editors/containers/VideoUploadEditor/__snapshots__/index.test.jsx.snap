// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoUploadEditor renders as expected with default behavior 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex marked-area flex-column p-3"
      >
        <div
          class="d-flex flex-column"
        >
          <div
            class="d-flex justify-content-end flex-row"
          >
            <button
              aria-label="Close"
              class="btn-icon btn-icon-primary btn-icon-md position-absolute mr-2 mt-2"
              type="button"
            >
              <span
                class="btn-icon__icon-container"
              >
                <span
                  class="pgn__icon btn-icon__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pgn__dropzone"
            data-testid="dropzone-container"
            role="presentation"
            tabindex="0"
          >
            <input
              accept="video/*,.mp4,.mov"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="d-flex flex-column justify-content-around align-items-center w-100"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="justify-content-center align-self-center rounded-circle bg-light-300 p-2.5"
                >
                  <span
                    class="pgn__icon text-muted"
                    style="height: 2rem; width: 2rem;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5 10h4v6h6v-6h4l-7-7-7 7zm0 8v2h14v-2H5z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="d-flex align-self-center justify-content-center flex-wrap flex-column pt-3"
                >
                  <span>
                    Drag and drop video here or click to upload
                  </span>
                  <span
                    class="x-small align-self-center pt-2"
                  >
                    Upload MP4 or MOV files (5 GB max)
                  </span>
                </div>
                <div
                  class="small align-self-center justify-content-center mx-2 text-dark font-weight-normal pt-3"
                >
                  OR
                </div>
                <div
                  class="zindex-9 video-id-prompt py-3"
                >
                  <div
                    class="input-group"
                  >
                    <div
                      class="pgn__form-control-decorator-group has-appended-node has-trailing-element m-0"
                    >
                      <input
                        aria-describedby="basic-addon2"
                        aria-label="Paste your video ID or URL"
                        class="form-control"
                        placeholder="Paste your video ID or URL"
                        value=""
                      />
                      <div
                        class="pgn__form-control-decorator pgn__form-control-decorator-trailing"
                      >
                        <button
                          aria-label="Submit"
                          class="btn-icon btn-icon-primary btn-icon-inline url-submit-button"
                          type="button"
                        >
                          <span
                            class="btn-icon__icon-container"
                          >
                            <span
                              class="pgn__icon btn-icon__icon"
                            >
                              <svg
                                aria-hidden="true"
                                fill="none"
                                focusable="false"
                                height="24"
                                role="img"
                                viewBox="0 0 24 24"
                                width="24"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8Z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex marked-area flex-column p-3"
    >
      <div
        class="d-flex flex-column"
      >
        <div
          class="d-flex justify-content-end flex-row"
        >
          <button
            aria-label="Close"
            class="btn-icon btn-icon-primary btn-icon-md position-absolute mr-2 mt-2"
            type="button"
          >
            <span
              class="btn-icon__icon-container"
            >
              <span
                class="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
        <div
          class="pgn__dropzone"
          data-testid="dropzone-container"
          role="presentation"
          tabindex="0"
        >
          <input
            accept="video/*,.mp4,.mov"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div
            class="d-flex flex-column justify-content-around align-items-center w-100"
          >
            <div
              class="d-flex flex-column"
            >
              <div
                class="justify-content-center align-self-center rounded-circle bg-light-300 p-2.5"
              >
                <span
                  class="pgn__icon text-muted"
                  style="height: 2rem; width: 2rem;"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 10h4v6h6v-6h4l-7-7-7 7zm0 8v2h14v-2H5z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="d-flex align-self-center justify-content-center flex-wrap flex-column pt-3"
              >
                <span>
                  Drag and drop video here or click to upload
                </span>
                <span
                  class="x-small align-self-center pt-2"
                >
                  Upload MP4 or MOV files (5 GB max)
                </span>
              </div>
              <div
                class="small align-self-center justify-content-center mx-2 text-dark font-weight-normal pt-3"
              >
                OR
              </div>
              <div
                class="zindex-9 video-id-prompt py-3"
              >
                <div
                  class="input-group"
                >
                  <div
                    class="pgn__form-control-decorator-group has-appended-node has-trailing-element m-0"
                  >
                    <input
                      aria-describedby="basic-addon2"
                      aria-label="Paste your video ID or URL"
                      class="form-control"
                      placeholder="Paste your video ID or URL"
                      value=""
                    />
                    <div
                      class="pgn__form-control-decorator pgn__form-control-decorator-trailing"
                    >
                      <button
                        aria-label="Submit"
                        class="btn-icon btn-icon-primary btn-icon-inline url-submit-button"
                        type="button"
                      >
                        <span
                          class="btn-icon__icon-container"
                        >
                          <span
                            class="pgn__icon btn-icon__icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="none"
                              focusable="false"
                              height="24"
                              role="img"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8Z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
